sbncode/BeamSpillInfoRetriever/BNBEXTRetriever/BNBEXTRetriever_module.cc

```diff
@@ -135,21 +149,38 @@ void sbn::EXTRetriever::produce(art::Event& e)
 } //end loop over events
 
 
-void sbn::EXTRetriever::beginSubRun(art::SubRun& sr)
+void sbn::BNBEXTRetriever::beginSubRun(art::SubRun& sr)
 {
+  TotalEXTCounts = 0;
+  totalMinBias = 0;
+  evtCount = 0;
+  scale_factor = 0;
   return;
 }
 
-//____________________________________________________________________________                                                                                                                                                                                      
-void sbn::EXTRetriever::endSubRun(art::SubRun& sr)
+//____________________________________________________________________________                                                                                                                                                                                     
+void sbn::BNBEXTRetriever::endSubRun(art::SubRun& sr)
 {
   // We will add all of the EXTCountInfo data-products to the 
   // art::SubRun so it persists 
+
+  if(evtCount != 0 && totalMinBias != 0)
+    scale_factor = 1. - (evtCount/totalMinBias);
```

Comment:
We need to scale by the minimum bias prescale rate, which can change run-to-run based on our configuration. The best way I found to do this was to use `frag.getDeltaGatesBNBOffMinbias()` BUT it doesn't work for event which haven't had a MinBias trigger yet.

Reply:
You can get the prescale rate with something like:
```cpp
void sbn::BNBExtRetriever::beginRun(art::Run& run) {
  auto const& trigConfig = run.getProduct<icarus::TriggerConfiguration>(fTrigConfigLabel);
  fPrescale = trigConfig.getMinBiasPrescale(sbn::triggerSource::OffbeamBNB);
}
```
(`#include "icaruscode/Decode/DataProducts/TriggerConfiguration.h"`)
With this information you can correct either statistically (`* (1.0 - 1.0/fPrescale)`) or exactly event by event.
The correct label from ICARUS `Stage0` processing is `triggerConfig` if I am not wrong.

I don't understand the meaning of the fraction `evtCount/totalMinBias`... I would understand a gate count instead of the triggered event count on the denominator as an estimation of the prescale factor.

I am not sure what the existing code does in the presence of events with `gate_type != 3`.